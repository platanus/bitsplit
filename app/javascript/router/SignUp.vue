<template>
  <div class="mt-16">
    <center>
      <div class="w-full max-w-xs">
        <div class="flex content-start mb-8">
          <router-link :to="signInRoute">
            <img :src="leftArrow" class="mr-2" align="top" />
          </router-link>
          <router-link :to="signInRoute">
            <p class="txt-field">
              Volver
            </p>
          </router-link>
        </div>
        <img :src="bitsplitLogo" class="mb-4" align="top" />

        <p class="txt-field mb-8">
          Crea tu cuenta, es fácil y rápido
        </p>
        <form @submit.prevent="handleSubmit" class="pt-4 pb-8 mb-4">
          <div class="mb-8">
            <textInput
              field-id="email"
              field-type="text"
              field-placeholder="Ingresa tu correo"
              field-name="email"
              v-model="email"
            />
          </div>
          <div class="mb-4">
            <passwordInput
              field-id="password"
              field-name="password"
              field-placeholder="Crea tu contraseña"
              v-model="password"
            />
            <passwordInput
              field-id="confirm_password"
              field-name="confirm_password"
              field-placeholder="Repite tu contraseña"
              v-model="confirm_password"
            />
          </div>
          <div class="mb-8">
            <checkBox
              field-text="Acepto los términos de privacidad"
              v-model="checkboxField"
            />
          </div>
          <submitButton :loading="signUpLoading" width="full">
            Crear Cuenta
          </submitButton>
        </form>
      </div>
    </center>
  </div>
</template>
<script>
import { mapActions, mapState } from 'vuex';
import textInput from '../components/Input';
import passwordInput from '../components/PasswordInput';
import submitButton from '../components/SubmitButton';
import checkBox from '../components/Checkbox';
import logo from '../assets/bitsplit-logo.svg';
import lArrow from '../assets/left-arrow.svg';

export default {
  name: 'SignUp',
  data() {
    return {
      routeName: 'SignUp',
      email: '',
      password: '',
      confirm_password: '',
      signInRoute: 'sign-in',
      checkboxField: false,
      bitsplitLogo: logo,
      leftArrow: lArrow,
    };
  },
  components: {
    textInput,
    passwordInput,
    submitButton,
    checkBox,
  },
  computed: {
    ...mapState('user', ['currentUser', 'signUpLoading']),
  },
  methods: {
    ...mapActions('user', ['signUp', 'signIn']),
    handleSubmit() {
      const { checkboxField, email, password, confirm_password } = this;
      if (checkboxField && email && password && password === confirm_password) {
        // TODO logger
        this.signUp({ email, password })
          .then(() =>
            // TODO logger
            this.signIn({ email, password })
          )
          .then(() => {
            this.$router.push('/onboarding');
          })
          .catch(err => {
            console.error(err);
          });
      }
    },
  },
};
</script>
