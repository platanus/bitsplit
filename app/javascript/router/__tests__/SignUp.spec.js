import { shallowMount } from '@vue/test-utils'
import SignUp from '../SignUp'

describe('SignUp', () => {
  it('matches snapshot', () => {
    const wrapper = shallowMount(SignUp)
    expect(wrapper).toMatchSnapshot()
  })
})
