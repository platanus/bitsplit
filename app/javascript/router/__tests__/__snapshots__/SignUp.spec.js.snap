// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp matches snapshot 1`] = `
<div class="mt-16">
  <center>
    <div class="w-full max-w-xs">
      <form class="bg-white shadow-md rounded px-8 pt-6 pb-8 mb-4">
        <div class="mb-4">
          <inputlabel-stub fieldfor="email">Email</inputlabel-stub>
          <textinput-stub value="" fieldid="email" fieldtype="text" fieldplaceholder="Email" fieldname="email" classmod=""></textinput-stub>
        </div>
        <div class="mb-4">
          <inputlabel-stub fieldfor="password">Contraseña</inputlabel-stub>
          <passwordinput-stub value="" fieldid="password" fieldname="password"></passwordinput-stub>
          <inputlabel-stub fieldfor="password">Confirme su contraseña</inputlabel-stub>
          <passwordinput-stub value="" fieldid="confirm_password" fieldname="confirm_password"></passwordinput-stub>
        </div>
        <submitbutton-stub classmod="bg-blue-500 hover:bg-blue-700">Registrarse</submitbutton-stub>
      </form>
    </div>
  </center>
</div>
`;
